File: src/WaterBottles.java
public class WaterBottles {
    public static void main(String[] args) {
        int numBottles = 9, numExchange = 3;

        System.out.println(numWaterBottles(numBottles, numExchange));
    }

    // ***************** 1st Method ******************
    // Approach 1: First set totalWaterBottles to numBottles, and add numBottles/numExchange to totalWaterBottles
    // Update numBottles to numBottles/numExchange + numBottles%numExchange
    // Runtime  : 0ms       -> + 100.00%
    // Memory   : 39.90MB   -> + 91.54%
    public static int numWaterBottles(int numBottles, int numExchange) {
        int totalWaterBottles = numBottles;

        while(numBottles >= numExchange) {
            totalWaterBottles += (numBottles/numExchange);
            numBottles = (numBottles/numExchange) + (numBottles%numExchange);
        }
        return totalWaterBottles;
    }
    // ***************** End of 1st Method ******************
}

File: CSharp-Solutions/LeetCodeCSharp/LeetCodeCSharp/WaterBottles.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LeetCodeCSharp
{
    public class WaterBottles
    {
        // ***************** 1st Method ******************
        // Runtime  : 13ms      -> + 100.00%
        // Memory   : 27.02MB   -> + 5.13%
        public int NumWaterBottles(int numBottles, int numExchange) {
            int totalBottles = numBottles;
            while(numBottles >= numExchange)
            {
                var curBottles = numBottles/numExchange;
                var leftover = numBottles%numExchange;
                totalBottles += curBottles;
                numBottles = leftover + curBottles;
            }
            return totalBottles;
        }
        // ***************** End of 1st Method ******************
    }
}